hypershift-operator/controllers/nodepool/inplace.go

```diff
@@ -100,6 +141,338 @@ func (r *NodePoolReconciler) reconcileInPlaceUpgrade(ctx context.Context, hc *hy
 	// Drain.
 	// Create Namespace/RBAC/ConfigMap/Pod in guest cluster.
 	// Mark Node as Upgrading.
+
+	// This check comes after the completion, so if no updates are in progress, if a node is degraded for
+	// whatever reason, we will not know until the next update, at which point hopefully the MCD is able
+	// to reconcile
+	for _, node := range nodes {
+		if node.Annotations[MachineConfigDaemonStateAnnotationKey] == MachineConfigDaemonStateDegraded {
+			setStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolCondition{
+				Type:               hyperv1.NodePoolUpdatingVersionConditionType,
```

Comment:
atm we make a difference between NodePoolUpdatingVersionConditionType and NodePoolUpdatingConfigConditionType. So will need to conflate both in one or keep the differentiation here. Adding a TODO for that here maybe ok for now.